pkg/apis/v1alpha1/vpcnetworkconfiguration_types.go

```diff
@@ -0,0 +1,115 @@
+/*
+Copyright 2022.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// SubnetType describes the subnet type.
+type SubnetType string
+
+const (
+	SubnetTypePublic  SubnetType = "public"
+	SubnetTypePrivate SubnetType = "private"
+)
+
+// VPCNetworkConfigurationSpec defines the desired state of VPCNetworkConfiguration
+type VPCNetworkConfigurationSpec struct {
+	// Namespaces to be applied
+	AppliedToNamespaces []string `json:"appliedToNamespaces,omitempty"`
+	// Tier0 gateway or Tier0 VRF path
+	// +kubebuilder:validation:Required
```

Comment:
This is related to my question for you on the design doc - if a field not set, it should inherit the Supervisor level setting or should use the default value? If we go the former strategy, then all fields should be optional. Could you check if VC API usually have optional fields or not? If yes, do they have default values or they can inherit settings of a higher level object (e.g. host inherits cluster settings)?

Reply:
Checked with Kevin that the VPC API can have optional values, but they do not have default values or inherit settings of a higher level object from the VMODL definition.  But in the implementation of the VAPI, we can set default values for optional fields if they are not set.

I think we can check with PM if we want the network settings on namespace inherit Supervisor level settings, like Edge cluster, public_ip_pools etc.